src/components/editMeasure/populationCriteria/QDMReporting/QDMReporting.test.tsx

```diff
@@ -0,0 +1,223 @@
+import * as React from "react";
+import { render, fireEvent, waitFor, screen } from "@testing-library/react";
+import { act } from "react-dom/test-utils";
+import useMeasureServiceApi, {
+  MeasureServiceApi,
+} from "../../../../api/useMeasureServiceApi";
+import { Measure } from "@madie/madie-models";
+import QDMReporting from "./QDMReporting";
+
+const mockHistoryPush = jest.fn();
+
+jest.mock("react-router-dom", () => ({
+  useHistory: () => ({
+    push: mockHistoryPush,
+  }),
+}));
+
+jest.mock("../../../../api/useMeasureServiceApi");
+const measure = {
+  id: "test measure",
+  measureName: "the measure for testing",
+  cqlLibraryName: "TestCqlLibraryName",
+  ecqmTitle: "ecqmTitle",
+  measurementPeriodStart: "01/01/2022",
+  measurementPeriodEnd: "12/02/2022",
+  createdBy: "john doe",
+  measureSetId: "testMeasureId",
+  acls: [{ userId: "othertestuser@example.com", roles: ["SHARED_WITH"] }], //#nosec
+} as unknown as Measure;
+
+jest.mock("@madie/madie-editor", () => ({
```

Comment:
Do we need to mock Editor?

Reply:
Jest does not understand anything that we're getting across the browser.

single spa is connected at the browser, so everything outside of your src or node_modules needs to be fully mocked like this.

Reply:
but we are only rendering the <QDMReporting /> component, and the editor will not even be a part of the rendered HTML in Jest.